src/Searchlight/LinqExecutor.cs

```diff
@@ -45,8 +45,35 @@ public static IEnumerable<T> QueryCollection<T>(this SyntaxTree tree, IEnumerabl
                     (from sort in tree.OrderBy select $"{sort.Column.FieldName} {sort.DirectionStr()}"));
                 queryable = queryable.OrderBy(sortExpression);
             }
+            
+            // If the user requested pagination
+            switch ((tree.PageNumber, tree.PageSize))
+            {
+                // case 1: user specified page number and page size
+                case (> 0, > 0):
+                    queryable = queryable.Skip((int) (tree.PageSize * tree.PageNumber)).Take((int) tree.PageSize);
+                    
+                    break;
+                
+                // case 2: user specified a page number but no page size
+                case (> 0, null):
+                    queryable = queryable.Skip((int) tree.PageNumber * 200).Take(200);
```

Comment:
This also doesn't seem right.  I don't think we have an assumption that "no page size means 200".

Reply:
I pulled it from the invoices, where default pagesize is 200.

Reply:
Searchlight isn't the same as Lockstep :) 

Let me file an issue for defining this at the engine level.

Reply:
I think the correct behavior here is to remove this use case.  If the user failed to specify pagination, and if the engine doesn't define a maximum page size, we should let them get all objects.

Reply:
Okay! I removed that case.